docs/DeveloperGuide.md

```diff
@@ -609,28 +610,294 @@ To sum it up, this application helps users to reduce the time needed to transfor
 
-## Appendix E: Instructions for Manual Testing
+**Note:** These instructions only provide a starting point for testers to work on, testers are expected to do more *exploratory* testing.
 
-### Start-up and Shutdown
+### Start and Shutdown
+
+1. Ensure you have Java version `11` or above installed on your computer.
+2. Next, download the latest `decodex.jar` [here](https://github.com/AY2122S1-CS2113T-T10-3/tp/releases).
+3. Copy the `decodex.jar` into an empty folder.
+4. Run the `decodex.jar` on your command prompt by typing in `java -jar decodex.jar` the press `Enter`.
+5. Type a command in the command box after the `Decodex [] >` prompt and press Enter to execute it. e.g. typing `help` and pressing `Enter` will show a list available commands, their respective descriptions and syntaxes.
+6. To exit Decodex, use the `exit` command. Alternatively, Decodex can be exited using `Ctrl+C` or by closing the command prompt window.
 
 ### Inputting Data
 
+When entering data, `Decodex` requires a specific format such that the data is entered correctly.
+
+<aside>
+ℹ️ - Note that the program will consider **all characters as data following the first space character** in the input command. You may observe leading and trailing whitespaces in the output.
+
+- **Non-printable characters** (e.g. null characters) may not be correctly displayed and could potentially scramble the user interface.
+
+- There will only be one instance of data at any time. Running the `input` command will overwrite the current data that is stored in the program.
+
+</aside>
+
+Here are some example inputs you can try:
+
+- input `hello world` : Inputs `hello world` as data into the Decodex.
+- input    `hello world`    : Inputs    `hello world`    as data into Decodex.
+- input `goodbye world` : Inputs `hello world` as data into the Decodex.
+
+For subsequent examples below, this guide assumes that the original input data is `goodbye world`.
+
 ### Listing Modules and Recipes
 
+The `list` command has 3 variations:
+
+- List all modules and recipes
+- List all modules
+- List all recipes
+
+The first command lists every single modules and recipes which is the combined output of the other 2 commands. No recipes will be found if no recipes have been created.
+
+- Test case `list`
+
+Expected (if no recipes exists): *do bullet point indent on github*
+
+```
+Here is the list of supported modules:
+  base64encode - Encodes the data using Base64 format.
+  base64decode - Decodes the data using Base64 format.
+  hexencode    - Converts the input string to hexadecimal bytes.
+  hexdecode    - Converts a hexadecimal string back into its raw value.
+  binencode    - Encodes the data using binary format.
+  bindecode    - Decodes the data using binary format.
+  rotencode    - Rotates alphabetical characters by a specified integer offset.
+
+You do not have any recipes
+```
+
+Expected (if some recipes exists):*do bullet point indent on github*
+
+```
```

Comment:
Missing bullet point indent

Reply:
Fixed but an additional bullet point is added as, in between the code blocks if only one tabbed indent is used, it appears as code block (the indented line)